login/service_whitebox_test.go

```diff
@@ -7,142 +7,194 @@ import (
 		}
 		if trials == 10 {
-			t.Error("Test failed, Maximum Retry limit reached", err) // Test failed after trial for 10 times
+			s.T().Error("Test failed, Maximum Retry limit reached", err) // Test failed after trial for 10 times
 		}
 	} else {
-		assert.NotEmpty(t, emails)
+		assert.NotEmpty(s.T(), emails)
 	}
 }
 
-func TestInvalidOAuthAccessToken(t *testing.T) {
-	resource.Require(t, resource.UnitTest)
-	setup()
-	defer tearDown()
-
-	if loginService == nil {
-		setup()
-	}
-
+func (s *serviceWhiteBoxTest) TestInvalidOAuthAccessToken() {
+	resource.Require(s.T(), resource.UnitTest)
 	invalidAccessToken := "7423742yuuiy-INVALID-73842342389h"
-
 	accessToken := &oauth2.Token{
 		AccessToken: invalidAccessToken,
 		TokenType:   "Bearer",
 	}
-
-	emails, err := loginService.getUserEmails(context.Background(), accessToken)
-	assert.Nil(t, err)
-	assert.Empty(t, emails)
+	emails, err := s.loginService.getUserEmails(context.Background(), accessToken)
+	assert.Nil(s.T(), err)
+	assert.Empty(s.T(), emails)
 }
 
-func TestGetUserEmails(t *testing.T) {
-	resource.Require(t, resource.UnitTest)
-
-	setup()
-	defer tearDown()
+func (s *serviceWhiteBoxTest) TestGetUserEmails() {
+	resource.Require(s.T(), resource.UnitTest)
+	accessToken := &oauth2.Token{
+		AccessToken: configuration.GetGithubAuthToken(),
+		TokenType:   "Bearer",
+	}
+	githubEmails, err := s.loginService.getUserEmails(context.Background(), accessToken)
+	s.T().Log(githubEmails)
+	assert.Nil(s.T(), err)
+	assert.NotNil(s.T(), githubEmails)
+	assert.NotEmpty(s.T(), githubEmails)
+}
 
+func (s *serviceWhiteBoxTest) TestGetUser() {
+	resource.Require(s.T(), resource.UnitTest)
 	accessToken := &oauth2.Token{
 		AccessToken: configuration.GetGithubAuthToken(),
 		TokenType:   "Bearer",
 	}
+	githubUser, err := s.loginService.getUser(context.Background(), accessToken)
+	assert.Nil(s.T(), err)
+	assert.NotNil(s.T(), githubUser)
+	s.T().Log(githubUser)
+}
 
-	githubEmails, err := loginService.getUserEmails(context.Background(), accessToken)
-	t.Log(githubEmails)
-	assert.Nil(t, err)
-	assert.NotNil(t, githubEmails)
-	assert.NotEmpty(t, githubEmails)
+func (s *serviceWhiteBoxTest) TestFilterPrimaryEmail() {
+	resource.Require(s.T(), resource.UnitTest)
+	s.T().Skip("Not implemented")
 }
 
-func TestGetUser(t *testing.T) {
-	resource.Require(t, resource.UnitTest)
-	setup()
-	defer tearDown()
+func (s *serviceWhiteBoxTest) TestCreateUser() {
+	resource.Require(s.T(), resource.Database)
+	defer gormsupport.DeleteCreatedEntities(s.DB)()
+	s.removeUserIfExists("sbose78@gmail.com")
 
 	accessToken := &oauth2.Token{
 		AccessToken: configuration.GetGithubAuthToken(),
 		TokenType:   "Bearer",
 	}
-
-	githubUser, err := loginService.getUser(context.Background(), accessToken)
-	assert.Nil(t, err)
-	assert.NotNil(t, githubUser)
-	t.Log(githubUser)
+	identity, err := s.loginService.getUserIdentity(context.Background(), accessToken)
+	if err != nil {
```

Comment:
Just make this `require.Nil(s.T(), err)` and your test will automatically fail. No need for the if logic.